src/traits.rs

```diff
@@ -6,3 +6,45 @@ pub trait ToGeo<T: Float>
 {
     fn to_geo(&self) -> Geometry<T>;
 }
+
+// FIXME: find good names for these traits, don't use XyzTrait naming scheme
+
+pub trait PointTrait<T: Float> {
+    fn x(&self) -> T;
+    fn y(&self) -> T;
+}
+
+pub trait LineStringTrait<T: Float> {
+    type ItemType: PointTrait<T>;
+    type Iter: Iterator<Item=Self::ItemType>;
+
+    fn points(&self) -> Self::Iter;
+}
+
+pub trait PolygonTrait<T: Float> {
+    type ItemType: LineStringTrait<T>;
+    type Iter: Iterator<Item=Self::ItemType>;
+
+    fn rings(&self) -> Self::Iter;
```

Comment:
Was going to discuss that eventually, so I guess we could do it now. For now, I'm just going to assume the first item in this iterator is the outer ring and following items are inner rings. We could probably migrate this to:

```rust
    fn outer_ring(&self) -> Self::ItemType;
    fn inner_rings(&self) -> Self::Iter {
        // Default is no inner rings
        std::iter::empty()
    }
```

Unless someone has a better idea for this.